artifacts/debian/debian.go

```diff
@@ -0,0 +1,163 @@
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/architecture"
+	"kubevirt.io/containerdisks/pkg/docs"
+	"kubevirt.io/containerdisks/pkg/http"
+	"kubevirt.io/containerdisks/pkg/tests"
+)
+
+type Annotations struct {
+	Digest string `json:"cloud.debian.org/digest"`
+}
+
+type ImageLabel struct {
+	ImageFormat string `json:"upload.cloud.debian.org/image-format"`
+	Version     string `json:"cloud.debian.org/version"`
+}
+
+type Metadata struct {
+	Annotations Annotations `json:"annotations"`
+	Labels      ImageLabel  `json:"labels"`
+}
+
+type Item struct {
+	Metadata Metadata `json:"metadata"`
+}
+
+type BuildData struct {
+	Items []Item `json:"items"`
+}
+
+type debian struct {
+	Arch            string
+	Version         string
+	VersionName     string
+	getter          http.Getter
+	ExampleUserData *docs.UserData
+	envVariables    map[string]string
+}
+
+const (
+	baseURLFmt  = "https://cloud.debian.org/images/cloud/%s/latest/"
+	baseNameFmt = "debian-%s-genericcloud-%s"
+	description = `Debian Generic Cloud images for KubeVirt.
+<br />
+<br />
+Visit [debian.org](https://cloud.debian.org/images/cloud/) to learn more about Debian project.`
+)
+
+func (d *debian) Metadata() *api.Metadata {
+	metadata := &api.Metadata{
+		Name:         "debian",
+		Version:      d.Version,
+		Description:  description,
+		Arch:         d.Arch,
+		EnvVariables: d.envVariables,
+	}
+
+	if d.ExampleUserData != nil {
+		metadata.ExampleUserData = *d.ExampleUserData
+	}
+
+	return metadata
+}
+
+func decodeChecksum(base64Checksum string) (string, error) {
+	base64Checksum = strings.TrimPrefix(base64Checksum, "sha512:")
+
+	rawBytes, err := base64.RawStdEncoding.DecodeString(base64Checksum)
+	if err != nil {
+		return "", fmt.Errorf("error decoding debian digest: %v", err)
+	}
+
+	return hex.EncodeToString(rawBytes), nil
+}
+
+func (d *debian) getBuildData(jsonURL string) (additionalTags []string, checksum string, err error) {
+	var buildData BuildData
+
+	raw, err := d.getter.GetAll(jsonURL)
+	if err != nil {
+		return []string{}, "", fmt.Errorf("error downloading debian json file: %v", err)
+	}
+	err = json.Unmarshal(raw, &buildData)
+	if err != nil {
+		return []string{}, "", fmt.Errorf("error decoding debian json file: %v", err)
+	}
+
+	if len(buildData.Items) == 0 {
+		return []string{}, "", fmt.Errorf("build debian data not found")
+	}
+
+	for _, item := range buildData.Items {
+		if item.Metadata.Labels.ImageFormat == "qcow2" {
+			additionalTags = append(additionalTags, d.Version+"-"+item.Metadata.Labels.Version)
+			checksum, err = decodeChecksum(item.Metadata.Annotations.Digest)
+			return additionalTags, checksum, err
+		}
+	}
+
+	return additionalTags, checksum, fmt.Errorf("error locating the image information")
+}
+
+func (d *debian) Inspect() (*api.ArtifactDetails, error) {
+	if !strings.HasPrefix(d.Version, "11") && !strings.HasSuffix(d.Version, "12") {
```

Comment:
Why are you mixing usage of `HasPrefix` and `HasSuffix`?

Reply:
Silly copy&paste mistake, `HasPrefix` is the right one.